static/js/stream_create.js

```diff
@@ -301,6 +301,7 @@ function clear_error_display() {
 export function show_new_stream_modal() {
     $("#stream-creation").removeClass("hide");
     $(".right .settings").hide();
+    stream_settings_ui.hide_or_disable_stream_privacy_options_if_required($("#stream-creation"));
```

Comment:
How does this state get reset if the settings change?  I don't see a show/inverse function, and I'm not sure that the stream creation panel gets removed and recreated in the DOM dynamically.

Reply:
I think after some manual testing, the live-update definitely doesn't work in either component if you change the policy without reloading the browser, so that needs to be debugged.  I'm OK with that being a little follow-up PR, so I don't need to re-review all the nice work in this PR.